examples/get.py

```diff
@@ -1,131 +1,93 @@
+                                              500,  # 500 Internal Server Error
+                                              502,  # 502 Bad Gateway
+                                              503,  # 503 Service Unavailable
+                                              504,  # 504 Gateway Timeout
+                                              ],
+                            )
+            a = HTTPAdapter(max_retries=retries)
+            session.mount('https://', a)
+
+        return session
+
+    def next_url(self):
+
+        for item in self.resp['data']:
+            self.resp_data['data'].append(item)
+
+        if args.page and self.resp['meta']['next']:
+            self.url = self.resp['meta']['next']
+            return self.url
+
+        if args.steps and self.resp['meta']['next']:
+            while args.steps != 0:
+                args.steps -= 1
+                self.url = self.resp['meta']['next']
+                return self.url
+
+    def get_data(self, get_url, json_output):
+
+        session = self.establish_session()
+
+        while get_url:
+
+            logging.info(get_url)
+
+            try:
+                r = session.get(get_url, headers=headers,
+                                timeout=30, stream=True)
 
-                        if len(self.resp['data']) < 1:
-                            return self.resp
+                if r.status_code != 200:
+                    logging.info(str(r.status_code) + ": " + str(r.text))
+                    break
 
-                        else:
-                            for item in self.resp['data']:
-                                resp_data['data'].append(item)
+                else:
+                    self.resp = json.loads(r.content.decode("utf-8"))
 
-                            if args.page and self.resp['meta']['next']:
-                                url = self.resp['meta']['next']
+                    if len(self.resp['data']) < 1:
+                        return self.resp
 
-                            if args.steps and self.resp['meta']['next']:
-                                while args.steps != 0:
-                                    args.steps -= 1
-                                    url = self.resp['meta']['next']
-                                    break
-                            else:
-                                url = None
+                    else:
+                        get_url = self.next_url()
 
-                except Exception as e:
-                    logging.info("Exception:", e)
-                    raise
+            except Exception as e:
+                logging.info("Exception:", e)
+                raise
 
-        data = json.dumps(resp_data, indent=4, sort_keys=False)
+        json_data = json.dumps(self.resp_data, indent=4, sort_keys=False)
 
-        with open(f'json/{output}', 'w') as outfile:
-            outfile.write(data)
+        with open(f'json/{json_output}', 'w') as outfile:
```

Comment:
`json.dump(resp_data, open(f`json/{output}', 'w'))`

Reply:
And let the caller of this function do the writing to file, anyway. This function is called `get_data`, not `get_data_and_by_the_way_write_it_to_this_file_while_you_are_at_it`.